tests/providers/test_adapter.py

```diff
@@ -240,17 +234,29 @@ def test_ionq_gates(self):
 
     def test_global_phase(self):
```

Comment:
This test iterates two test cases. `@pytest.mark.parametrize` may be more appropriate here.